diff --git a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.d.ts b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.d.ts
index 42bff4a..8d9e34e 100644
--- a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.d.ts
+++ b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.d.ts
@@ -43,4 +43,6 @@ export interface KeyboardAvoidingViewProps extends ViewProps {
    * Default is true
    */
   enabled?: boolean | undefined;
+
+  ignoreIOSKeyboardWillChangeEvents?: boolean | undefined;
 }
diff --git a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
index e26d677..2fb58b5 100644
--- a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
+++ b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
@@ -49,6 +49,8 @@ type Props = $ReadOnly<{|
    * may be non-zero in some cases. Defaults to 0.
    */
   keyboardVerticalOffset?: number,
+
+  ignoreIOSKeyboardWillChangeEvents?: boolean | undefined;
 |}>;
 
 type State = {|
@@ -146,6 +148,9 @@ class KeyboardAvoidingView extends React.Component<Props, State> {
     }
 
     const {duration, easing, endCoordinates} = this._keyboardEvent;
+    if(Platform.OS === 'ios' && duration === 0 && !!this.props.ignoreIOSKeyboardWillChangeEvents ) {
+      return;
+    }
     const height = await this._relativeKeyboardHeight(endCoordinates);
 
     if (this._bottom === height) {
